docs/permission/writing-a-policy.md

```diff
@@ -38,36 +38,50 @@ Let's change the policy to the following:
-+       return createCatalogPolicyDecision(
-+         catalogConditions.isEntityOwner(
-+           user?.identity.ownershipEntityRefs ?? [],
-+         ),
-+       );
-      }
++     if (isResourcePermission(request.permission, 'catalog-entity')) {
+        if (request.permission.name === 'catalog.entity.delete') {
+-         return {
+-           result: AuthorizeResult.DENY,
+-         };
++         return createCatalogConditionalDecision(
++           request.permission,
++           catalogConditions.isEntityOwner(
++             user?.identity.ownershipEntityRefs ?? [],
++           ),
++         );
+        }
++     }
 
       return { result: AuthorizeResult.ALLOW };
     }
   }
 ```
 
-Let's walk through the new code that we just added. Inside of the if statement, instead of returning a Definitive Policy Decision of DENY, we now return a [Conditional Policy Decision](https://backstage.io/docs/reference/plugin-permission-node.conditionalpolicydecision) (See the [Concepts page](./concepts.md) for more details). This is a way for policies to defer the evaulation of the decision back to the plugin which owns the permission. This allows the framework to support cases in which the policy does not have all the information required to make a decision.
+Let's walk through the new code that we just added. Inside of the if statement, instead of returning a Definitive Policy Decision of DENY, we now return a [Conditional Policy Decision](https://backstage.io/docs/reference/plugin-permission-common.conditionalpolicydecision) (See the [Concepts page](./concepts.md) for more details). This is a way for policies to defer the evaluation of the decision back to the plugin which owns the permission. This allows the framework to support cases in which the policy does not have all the information required to make a decision.
 
-In the policy above, there's no way for the handle method to determine whether the user who is trying to unregister the entity is the owner of that entity. So we use the `createCatalogPolicyDecision` helper provided by the catalog backend to craft a conditional decision, which allows us to tell the catalog backend that it should only return ALLOW if the user owns the entity.
+In the policy above, there's no way for the handle method to determine whether the user who is trying to unregister the entity is the owner of that entity. So we use the `createCatalogConditionalDecision` helper provided by the catalog backend to craft a conditional decision, which allows us to tell the catalog backend that it should only return ALLOW if the user owns the entity. Keep in mind that, in order to return a Conditional Policy Decision, the permission needs to be of type [ResourcePermission](https://backstage.io/docs/reference/plugin-permission-common.resourcepermission): the [isResourcePermission](https://backstage.io/docs/reference/plugin-permission-common.isresourcepermission) method guarantees the correct type in the scope of the condition.
```

Comment:
Also worth mentioning that it should be a `ResourcePermission` with the correct resource type. `createCatalogConditionalDecision` will only accept `ResourcePermission<'catalog-entity'>`